apple/internal/resource_actions/plist.bzl

```diff
@@ -240,6 +243,10 @@ def merge_root_infoplists(
         # been valid bugs, so this will still catch that.
         plists.append(struct(CFBundleIdentifier = bundle_id))
 
+    if bundle_package_type:
+        substitutions["PRODUCT_BUNDLE_PACKAGE_TYPE"] = bundle_package_type
+        plists.append(struct(CFBundlePackageType = bundle_package_type))
```

Comment:
One thing that's notable here that wasn't happening before is that this line will validate that `CFBundlePackageType` is set correctly for the current bundle. Not sure if there are any recommendations on whether this should be rolled out slowly with a flag or merged in its current state (where it may fail builds that have the wrong type set in their plists)

Reply:
yea I thought about this too, one interesting thing is that with resource bundles (which I guess should be `BNDL`) we currently don't set `CFBundlePackageType` at all, and everything works in prod, so I guess it's not always validated. So for us we will start setting this based on this

Reply:
As-is, if you don't have it set _at all_, this will set it for you. If you have it set incorrectly (or not specifically enough), the build will fail.